Steps4Impact/LoginV2/LoginV2ViewController.swift

```diff
@@ -277,8 +278,28 @@ extension LoginV2ViewController {
   
   @objc
   private func googleLoginButtonTapped() {
-    GIDSignIn.sharedInstance()?.presentingViewController = self
-    GIDSignIn.sharedInstance()?.signIn()
+    let config = GIDConfiguration(clientID: FirebaseApp.app()?.options.clientID ?? "")
+    GIDSignIn.sharedInstance.signIn(with: config, presenting: self) { user, error in
+      guard error == nil else {
+        NotificationCenter.default.post(name: .googleSigninFailure, object: error)
+        return
+      }
+      
+      guard let authentication = user?.authentication else {
+        // This shouldn't happen if error == nil and we don't have a way to recover except try again
+        NotificationCenter.default.post(name: .googleSigninFailure, object: "Unknown Error")
+        return
+      }
+      
+      guard let idToken = authentication.idToken else {
+        NotificationCenter.default.post(name: .googleSigninFailure, object: "Unknown Error")
```

Comment:
Should we differentiate between L290 and L295?

Reply:
Wasn't able to find a user facing error message for missing idToken. I'm assuming that either both authentication and idToken will be missing, or neither. Updated to (per: https://firebase.google.com/docs/auth/ios/google-signin):
```swift
guard
        let authentication = user?.authentication,
        let idToken = authentication.idToken
      else {
```